### PR TITLE
configuration: optionally read pending configs in migration

### PR DESCRIPTION
**Pull request by slumber**
Follows up #7396 

CI was failing because of warning caused in migration. The warning is coming from `defensive_proof` call on optional value of `PendingConfigs`, which is actually allowed to be `None`. My bad for blindly copying code from https://github.com/paritytech/polkadot/issues/6271.

### CODE REVIEW

File: runtime/parachains/src/configuration/migration/v7.rs [hunk 1]
```diff
@@ -156,9 +156,8 @@ executor_params                          : pre.executor_params,
 	let v7 = translate(v6);
 	v7::ActiveConfig::<T>::set(Some(v7));
 
-	let pending_v6 = v6::PendingConfigs::<T>::get()
-		.defensive_proof("Could not decode old pending")
-		.unwrap_or_default();
+	// Allowed to be empty.
+	let pending_v6 = v6::PendingConfigs::<T>::get().unwrap_or_default();
```
[the remark this answers is not in the document]
Comment (slumber) [reply runtime/parachains/src/configuration/migration/v7.rs:160]: Doesn't `StorageValue::get` already account for undecodable value? Besides, `decode_len` would still panic with `defensive`.

**Comment by slumber** [comment]
Despite the fact that the warning is gone, there's still `Corrupted state ...`.

**Comment by slumber** [comment]
### Reasoning behind failing CI:

https://github.com/paritytech/polkadot/blob/master/runtime/parachains/src/configuration/migration_ump.rs
This migration schedules config upgrade for a set of values, without altering the struct.

Note that this migration does it based on _current_ code of host config, the one we're supposed to have after applying new (v7) migration.

It adds another item to `PendingConfigs`, of version 7, when trying to upgrade it from v6 to v7 we fail with "Corrupted state".

### Why is this harmless?

We can't skip this migration based on pallet storage version, it's supposed to be applied for v6, the same way my config migration is supposed to be applied for v6.
UMP migration is already part of release, and it's not guarded against being re-executed. However, since we fail to read pending config upgrades, ump migration will be discarded with execution of v7 (we failed to read it).
Assuming it was previously applied, this is ok.

### Solution?

Linked migration was supposed to bump storage version (unfortunately it didn't). Then we could guard against re-executing it, thus mitigating read-error. However, altering migration that was already applied sounds bad to me, even though it could work in theory.

So unless V0943 migrations get dropped, only option is to ignore this warning.

**Comment by ordian** [comment]
> Assuming it was previously applied, this is ok.

UMP migration was not applied on Polkadot yet:
https://github.com/paritytech/polkadot/blob/aa5c8220ff6705f040f44d06479dd88b8e7e4601/runtime/polkadot/src/lib.rs#L1560-L1563

Screenshot from 2023-07-11 17-46-22

But it probably will be by the time v7 hits the runtime.

**Comment by ggwpez** [comment]
Polkadot should upgrade to .43 in 16 days or so?  
So you can probably already remove the .43 migrations (and older) to fix this. I have to admit that this `UmpLimits` "migration" was very hacky, but the limits needed to be bumped synchronously with the runtime upgrade for security reasons...

**Review by ggwpez — APPROVED** [review]

**Comment by ordian** [comment]
Why are runtime migration checks are failing btw? Is that expected? I don't see them failing on other PRs.

**Comment by slumber** [comment]
> Polkadot should upgrade to .43 in 16 days or so? So you can probably already remove the .43 migrations (and older) to fix this. I have to admit that this `UmpLimits` "migration" was very hacky, but the limits needed to be bumped synchronously with the runtime upgrade for security reasons...

We can do this in a follow-up to address the warning.

**Comment by slumber** [comment]
> Why are runtime migration checks are failing btw? Is that expected? I don't see them failing on other PRs.

https://github.com/paritytech/polkadot/pull/7489#issuecomment-1630976076 is exact explanation. I've merged config migration very recently so there're no PRs built on top of it

**Comment by ggwpez** [comment]
> We can do this in a follow-up to address the warning.

Then we have to ensure that this does not get into a release. I am not sure if the runtime 1.0.0 release is already branched off.

**Comment by slumber** [comment]
> > We can do this in a follow-up to address the warning.
> 
> Then we have to ensure that this does not get into a release. I am not sure if the runtime 1.0.0 release is already branched off.

It's safe to be released as long as we drop earlier migrations? I meant that this PR addresses unrelated thing and don't want to include these changes here.

**Review by ordian — APPROVED** [review]

**Comment by ordian** [comment]
bot merge

**Comment by ordian** [comment]
> > Polkadot should upgrade to .43 in 16 days or so? So you can probably already remove the .43 migrations (and older) to fix this. I have to admit that this `UmpLimits` "migration" was very hacky, but the limits needed to be bumped synchronously with the runtime upgrade for security reasons...
> 
> We can do this in a follow-up to address the warning.

Please do. 

> > Why are runtime migration checks are failing btw? Is that expected? I don't see them failing on other PRs.
> 
> [#7489 (comment)](https://github.com/paritytech/polkadot/pull/7489#issuecomment-1630976076) is exact explanation. I've merged config migration very recently so there're no PRs built on top of it

Yeah, sorry, I misinterpreted the explanation to be for the previous PR.

**Pull request merged**